src/snowflake/snowpark/mock/_functions.py

```diff
@@ -1027,7 +1025,7 @@ def convert_char(row):
             )
 
     # row index information is needed to retrieve format information in another pd series, thus calling to_frame here
-    res = column.to_frame().apply(convert_char, axis=1)
+    res = column.combine(fmt, convert_char)
```

Comment:
_to_timestamp implementation also uses `column.to_frame()`, should we fix that as well?
I'm good with either fixing in this PR or having another PR.

Reply:
I have created https://github.com/snowflakedb/snowpark-python/issues/1740.